### ISSUE TITLE
carthage bootstrap fetching sub-dependencies when it doesn't need to

### ISSUE THREAD

**Issue by BobElDevil** [Carthage/Carthage#1041]
Noticed while working on #1032. Running the following with a completely up to date repository cache shows an issue:
```
$ cat Cartfile.resolved                 
github "robrix/Box" "1.2.2"
github "antitypical/Result" "0.4.5"
github "ReactiveCocoa/ReactiveCocoa" "v3.0.0"
$ carthage bootstrap --no-build  
*** Fetching Result
*** Fetching Box
*** Checking out Box at "1.2.2"
*** Checking out Result at "0.4.5"
*** Checking out ReactiveCocoa at "v3.0.0"
```

Even though everything is up to date, we're still fetching Result & Box. The changes to make the Cartfile.resolved order stable involved running the dependency resolver even for bootstrap. However with an up to date cache, we shouldn't need to fetch anything, as we already know exactly what tags/commits we should be using. 

A possible solution would be to provide the Resolver with a set of already determined commits/tags for a dependency. When it's resolving what to build next, if the dependencies being analyzed already have versions specified, just use that version immediately, by-passing the meat of the version resolution logic.

**Comment by mdiep** [comment]
> The changes to make the Cartfile.resolved order stable involved running the dependency resolver even for bootstrap

I'd like to rework how this is done so that it's outside of `Resolver`. I'm not sure whether that'd solve the larger issue here or not, but I think this functionality is unnecessarily coupled to `Resolver`.

**Comment by ikesyo** [comment]
This was addressed by #1100.
